addons/account/views/account_move_views.xml

```diff
@@ -61,7 +61,7 @@
                                     </group>
                                     <group string="Currency" groups="base.group_multi_currency">
                                         <field name="currency_id" invisible="1"/>
-                                        <field name="amount_currency"/>
+                                        <field name="amount_currency" readonly="1" />
```

Comment:
It can still be editable for tax lines (`display_type == "tax"`) but not if the journal entry is posted (so editable only for tax lines of draft moves)